examples/friendly-pirate-app/scenarios.js

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2018 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+const {
+  describe, it, get,
+} = require('parrot-friendly');
+// const graphql = require('parrot-graphql');
+
+// const schema = `
+// type Query {
+//   shiplog: [ShipLog]
+// }
+
+// type ShipLog {
+//   Name: String!
+//   Captain: String!
+// }
+// `;
+
+const scenarios = describe('Scenarios (parrot-friendly)', () => {
+  it('has one friendly ship', () => {
+    get('/ship_log').response([{ name: 'The Jolly Roger', captain: 'Captain Hook' }]);
+  });
+
+  it('has more friendly ships', () => {
+    get('/ship_log').response([
+      { name: 'The Jolly Roger', captain: 'Captain Hook' },
+      { name: 'The Black Pearl', captain: 'Jack Sparrow' },
+      { name: 'Flying Dutchman', captain: 'Davy Jones' },
+      { name: 'The Wanderer', captain: 'Captain Ron' },
+    ]);
+  });
+
+  it('has a partial success and an error', () => {
+    get('/ship_log').data([
+      { name: 'The Jolly Roger', captain: 'Captain Hook' },
+      { name: 'The Black Pearl', captain: 'Jack Sparrow' },
+      { name: 'The Wanderer', captain: 'Captain Ron' },
+    ]).errors([
+      { message: "We're missing Davey Jones!" }
+    ]);
+  });
+
+  it('has a partial success and multiple errors', () => {
+    get('/ship_log').data([
+      { name: 'The Jolly Roger', captain: 'Captain Hook' },
+      { name: 'The Black Pearl', captain: 'Jack Sparrow' },
+    ]).errors([
+      { message: "We're missing Davey Jones!" },
+      { message: "We're missing Captain Ron!" }
+    ]);
+  });
+
+  // it('has full graphql success', () => {
+  //   // graphql('/graphql', schema, {
+  //   //   Query: () => ({
+  //   //     Name: () => 'Jolly Roger',
+  //   //     Captain: () => 'Black Beard',
+  //   //   })
+  //   // })
+  //   graphql('/graphql', schema, () => ({
+  //     ShipLog: () => { data: 'response' }
+  //   }))
+  // });
```

Comment:
I would appreciate some help with this mock, I want to prove that a graphql partial response is working as intended but was running into proxy issues.

Reply:
Can you explain the issues you were having?